sycl/source/detail/graph_impl.hpp

```diff
@@ -376,6 +620,33 @@ class exec_graph_impl {
     return MSchedule;
   }
 
+  /// Prints the contents of the graph to a text file in DOT format
+  /// @param GraphName is a string appended to the output file name
+  void printGraphAsDot(const char *GraphName) {
```

Comment:
Would prefer to pass a `std::string` rather than a C string in the parameter, don't need to worry about a NULL pointer or non `\0` terminated input. We're doing this explicitly later anyway in `std::string GraphNameStr(GraphName);`